tests/core/full_node/test_full_node.py

```diff
@@ -171,9 +171,13 @@ async def check_transaction_confirmed(transaction) -> bool:
         if consensus_mode == ConsensusMode.HARD_FORK_2_0:
             # after the hard fork we don't use this compression mechanism
             # anymore, we use CLVM backrefs in the encoding instead
-            assert detect_potential_template_generator(uint32(5), program) is None
+            check = lambda x: x is None  # noqa: 731
+            check_num_blocks = lambda x: x == 0  # noqa: 731
         else:
-            assert detect_potential_template_generator(uint32(5), program) is not None
+            check = lambda x: x is not None  # noqa: 731
+            check_num_blocks = lambda x: x > 0  # noqa: 731
+        assert check(detect_potential_template_generator(uint32(5), program))
```

Comment:
Yeah, I did that, and then I kept going because it still seemed quite verbose. I was surprised to learn that you can't have `assert` in a lambda, that was weird.
But sure, I'll undo the lambdas

Reply:
Lambda is just syntax and that syntax only allows a single expression, so no statements like `assert`.  I mean you can cheat and do stuff like `lambda x: [print(x), 2 * x][1]` to 'do multiple things and return one value' but...